swagger_to/style.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python3
+"""
+Generates code for an Elm client.
+"""
+# pylint: disable=too-many-lines
+from typing import MutableSet, MutableMapping, List
+
+import swagger_to.intermediate
+import swagger_to.swagger
+import swagger_to
+
+
+def check_header(swagger: swagger_to.swagger) -> List[str]:
+    """
+    Checks whether the swagger header conforms to the style guide.
+
+    :param swagger: parsed swagger file
+    :return: the list of failed checks
+    """
+    errs = []  # type: List[str]
+
+    if swagger.name != swagger_to.snake_case(swagger.name):
+        errs.append("Swagger name {} is not snake case.".format(swagger.name))
+
+    if not swagger.base_path.startswith("/"):
+        errs.append("Swagger base path {} doesn't start with a slash.".format(swagger.base_path))
+
+    if swagger.description.capitalize() != swagger.description:
+        errs.append("Swagger description \"{}\" should be capitalized")
+
+    return errs
+
+
+def check_cases(typedefs: MutableMapping[str, swagger_to.intermediate.Typedef],
```

Comment:
I'd split this function in two (`check_casing_typedefs`, `check_casing_endpoints`) for readability.